### PR TITLE
Add `test.sh` to run Python app tests

### PR DESCRIPTION
**Pull request by Yo1k**
Also add `clean.sh`, add description to scripts.

### CODE REVIEW

File: clean.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+# Cleans the build directory, that consists of Python app packages for different platforms.
```
Comment (stIncMale) [inline clean.sh]: I am not sure _"that consists of Python app packages for different platforms"_ is a useful clarification. It also has a chance to become outdated in the future.

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+# Runs all unit tests of the Python app.
```
Comment (stIncMale) [inline test.sh]: The _"of the Python app"_ part does not seem to be useful. It is also worth pointing out that the unittest framework may be also used to write and run not only unit tests, but also, e.g., integration tests. I propose the "Runs all tests" description.

File: activate-venv.sh [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# Activates Python project's virtual environment.
```
Comment (stIncMale) [inline activate-venv.sh]: Is it just me, or does _"Activates Python virtual environment for the project"_ sound better? It appears that you try to emphasize that Yo1k/tic-tac-toe is a Python project / Python application. This emphasis does not seem to be useful.

File: harness-util/util.sh [hunk 4]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env bash
+# Contains global variables.
```
Comment (stIncMale) [inline harness-util/util.sh]: These comments are not meant to be literal, they are meant to give a quick understanding of the purpose of the script.

P.S. This review comment is also relevant to `package-util.sh` and `python-util.sh`.

File: harness-util/package-util.sh [hunk 5]
```diff
@@ -50,3 +53,7 @@ function create_pkg() {
     install_python_requirements "${1}" "${2}" "${4}"
     pack "${1}" "${2}"
 }
+
+function clean_build(){
```
Comment (stIncMale) [inline harness-util/package-util.sh]: 1. `clean_build` -> `clean_build_dir`?
2. There seems to be a space missing in `(){`.

File: package_linux.sh [hunk 6]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# Packages the Python app with all its dependencies for Linux x86-64 platform.
```
Comment (stIncMale) [inline package_linux.sh]: _"Packages the Python app"_ -> _"Packages the app"_ or _"Packages the application"_? The same is relevant for `package_macos.sh`, `package_windows.sh`.

File: tic-tac-toe.sh [hunk 7]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# Runs the Python app on Linux or macOS.
```
Comment (stIncMale) [inline tic-tac-toe.sh]: 1. The script is agnostic to the OS. It is written for a specific shell (Bash), not a specific OS.
2. Consider using the wording _"An application launcher for Bash"_.
3. I propose to stop emphasizing that this is a Python app in places where this fact is irrelevant.

File: tic-tac-toe.cmd [hunk 8]
```diff
@@ -1,2 +1,4 @@
+REM Runs the Python app on Windows.
```
Comment (stIncMale) [inline tic-tac-toe.cmd]: Similarly to what I proposed in `tic-tac-toe.sh`, consider using the wording _"An application launcher for Windows Command shell and PowerShell"_. Note that I did not invent the name "Windows Command" for the built-in shell, it's how the shell is actually called in the docs: https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/windows-commands.

File: update-pip-requirements.sh [hunk 9]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# Updates `requirements.txt`, that stores all Python project's dependencies.
```
Comment (stIncMale) [inline update-pip-requirements.sh]: Consider
```
# Updates all Python dependencies of the project in `requirements.txt` based on the project's Python virtual environment.
# In order to add a new dependency / upgrade the existing dependency:
# - install / upgrade the dependency in the project's Python virtual environment;
# - run this script.
```

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: clean.sh (hunk 1 above)
Comment (Yo1k) [reply clean.sh]: Done.

File: test.sh (hunk 2 above)
Comment (Yo1k) [reply test.sh]: Done.

File: activate-venv.sh (hunk 3 above)
Comment (Yo1k) [reply activate-venv.sh]: Done.

File: harness-util/util.sh (hunk 4 above)
Comment (Yo1k) [reply harness-util/util.sh]: Done for all three files: `util.sh`, `package-util.sh`, and `python-util.sh`.

File: harness-util/package-util.sh (hunk 5 above)
Comment (Yo1k) [reply harness-util/package-util.sh]: Done.

File: package_linux.sh (hunk 6 above)
Comment (Yo1k) [reply package_linux.sh]: Done.

File: tic-tac-toe.sh (hunk 7 above)
Comment (Yo1k) [reply tic-tac-toe.sh]: Done.

File: tic-tac-toe.cmd (hunk 8 above)
Comment (Yo1k) [reply tic-tac-toe.cmd]: Done.

File: update-pip-requirements.sh (hunk 9 above)
Comment (Yo1k) [reply update-pip-requirements.sh]: I have accepted this suggestion.

File: harness-util/package-util.sh (hunk 5 above)
Comment (stIncMale) [reply harness-util/package-util.sh]: The space is still missing.

Comment (Yo1k) [reply harness-util/package-util.sh]: Done.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**